pkg/github/github_manager.go

```diff
@@ -0,0 +1,71 @@
+package github
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/ghodss/yaml"
+	klum "github.com/jadolg/klum/pkg/apis/klum.cattle.io/v1alpha1"
+	log "github.com/sirupsen/logrus"
+)
+
+func UploadGithubSecret(kubeconfig *klum.Kubeconfig, user *klum.User, githubURL string, githubToken string) error {
+	if user.Spec.Sync.Github == (klum.GithubSyncSpec{}) {
+		return nil
+	}
+	if user.Spec.Sync.Github.SecretName == "" ||
+		user.Spec.Sync.Github.Owner == "" ||
+		user.Spec.Sync.Github.Repository == "" {
+		return fmt.Errorf("not enough github data to be able to create a GitHub secret")
+	}
+
+	log.Infof("Adding secret (%s) to GitHub for user %s to %s/%s", user.Spec.Sync.Github.SecretName, kubeconfig.Name, user.Spec.Sync.Github.Owner, user.Spec.Sync.Github.Repository)
+
+	kubeconfigYAML, err := toYAMLString(kubeconfig.Spec)
+	if err != nil {
+		return err
+	}
+
+	return createRepositorySecret(
+		githubURL,
+		user.Spec.Sync.Github.Owner,
+		user.Spec.Sync.Github.Repository,
+		user.Spec.Sync.Github.Environment,
+		user.Spec.Sync.Github.SecretName,
+		kubeconfigYAML,
+		githubToken,
+	)
+}
+
+func DeleteGithubSecret(user *klum.User, githubURL string, githubToken string) error {
+	if user.Spec.Sync.Github.SecretName == "" ||
+		user.Spec.Sync.Github.Owner == "" ||
+		user.Spec.Sync.Github.Repository == "" {
+		log.Info("Not enough github data to be able to remove a GitHub secret")
+		return nil
+	}
```

Comment:
Could this be checked in a validating webhook?

Reply:
I'm too new to this framework and same as with the tests, I have no idea how to add the validation webhook. It's one of the things I plan to add in the future.

Reply:
did a validation here [db6847a](https://github.com/jadolg/klum/pull/19/commits/db6847ac3e227dfbfac491ace09cf4c34a2a1332)

Reply:
argument.Valid == true

Reply:
nah, the 3 together.
I'll improve this from the validation webhook later.